### PR TITLE
minimal-bootstrap: init `gnutar`, `gzip`, `bzip2`

### PR DESCRIPTION
**Pull request by emilytrau**
###### Description of changes

Adds the essential archive tools. After this we will have a complete set of replacements for `mescc-tools-extra`.

Related #227914

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/minimal-bootstrap/gzip/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  inherit pname version;
+
+  nativeBuildInputs = [
+    tinycc.compiler
+    gnumake
+    gnused
+    gnugrep
+  ];
+
+  passthru.tests.get-version = result:
```
Comment (06kellyjac) [inline pkgs/os-specific/linux/minimal-bootstrap/gzip/default.nix:28]: since these now take a `result` we need to feed that in for the test now right?

Comment (emilytrau) [reply pkgs/os-specific/linux/minimal-bootstrap/gzip/default.nix:28]: It's automatically fed in from `derivationWithMeta` :D
https://github.com/NixOS/nixpkgs/blob/4aa442cab08366eeec3945a46ffa8e67449649b8/pkgs/os-specific/linux/minimal-bootstrap/utils.nix#L26

Comment (06kellyjac) [reply pkgs/os-specific/linux/minimal-bootstrap/gzip/default.nix:28]: ahh. so does everything in `passthru.tests` need to be a function that takes self/result?

Comment (Ericson2314) [reply pkgs/os-specific/linux/minimal-bootstrap/gzip/default.nix:28]: Just for these bootstrapping things. `mkDerivation` does it differently with `finalAttrs: { ... }`.

**Review by Ericson2314 — APPROVED** [review]

**Pull request merged**